src/main/java/gregtech/api/capability/impl/MultiblockRecipeLogic.java

```diff
@@ -368,32 +392,35 @@ protected boolean drawEnergy(int recipeEUt, boolean simulate) {
     }
 
     @Override
-    protected long getMaxVoltage() {
+    public long getMaxVoltage() {
         IEnergyContainer energyContainer = getEnergyContainer();
-        if (energyContainer instanceof EnergyContainerList) {
-            long voltage;
-            long amperage;
-            if (energyContainer.getInputVoltage() > energyContainer.getOutputVoltage()) {
-                voltage = energyContainer.getInputVoltage();
-                amperage = energyContainer.getInputAmperage();
-            } else {
-                voltage = energyContainer.getOutputVoltage();
-                amperage = energyContainer.getOutputAmperage();
-            }
-
-            if (amperage == 1) {
-                // amperage is 1 when the energy is not exactly on a tier
-
-                // the voltage for recipe search is always on tier, so take the closest lower tier
+        if (!consumesEnergy()) {
+            // Generators
+            long voltage = energyContainer.getOutputVoltage();
+            long amperage = energyContainer.getOutputAmperage();
+            if (energyContainer instanceof EnergyContainerList && amperage == 1) {
+                // Amperage is 1 when the energy is not exactly on a tier.
+                // The voltage for recipe search is always on tier, so take the closest lower tier.
+                // List check is done because single hatches will always be a "clean voltage," no need
+                // for any additional checks.
                 return GTValues.V[GTUtility.getFloorTierByVoltage(voltage)];
+            }
+            return voltage;
+        } else {
+            // Machines
+            if (energyContainer instanceof EnergyContainerList energyList) {
+                long highestVoltage = energyList.getHighestInputVoltage();
+                if (energyList.getNumHighestInputContainers() > 1) {
+                    // allow tier + 1 if there are multiple hatches present at the highest tier
+                    int tier = GTUtility.getTierByVoltage(highestVoltage);
+                    return GTValues.V[Math.min(tier + 1, GTValues.MAX)];
```

Comment:
Should this use `tier + (energyList.getNumHighestInputContainers() - 1)` instead of `tier + 1`?

Reply:
Well, this code is assuming that you can never have enough hatches to go up that many tiers. But if we want to account for that, then it should be `tier + log2(energyList.getNumHighestInputContainers())`